src/hotspot/share/jvmci/jvmciCompilerToVM.cpp

```diff
@@ -1000,7 +1000,7 @@ C2V_VMENTRY_NULL(jobject, resolveFieldInPool, (JNIEnv* env, jobject, ARGUMENT_PA
   if (info.is_null() || JVMCIENV->get_length(info) != 4) {
     JVMCI_ERROR_NULL("info must not be null and have a length of 4");
   }
-  JVMCIENV->put_int_at(info, 0, fd.access_flags().as_int());
+  JVMCIENV->put_int_at(info, 0, fd.access_flags().as_unsigned_short());
```

Comment:
Again are we relying on implicit widening from u2 to int?

It really isn't clear to me whether the only thing we should have changed here is the actual type of the `_flags` field and let everything else continue to represent flags as int, so we don't get these transitions from u2 to int  in these higher level APIs.

Reply:
Yes, we can widen u2 to int.  See above.  The ci code represents the integral value of access flags as jint.  I am leaving that API in place.  For this, the widening happens when fetching the u2 field.  The conversion is implicit.  If this field were to be stored back to a u2 somewhere, all the code in the compiler should change but the current code doesn't do that.